test/Npgsql.PluginTests/NetTopologySuiteTests.cs

```diff
@@ -0,0 +1,225 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2018 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using GeoAPI;
+using GeoAPI.Geometries;
+using NetTopologySuite.Geometries;
+using Npgsql.Tests;
+using NUnit.Framework;
+using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+namespace Npgsql.PluginTests
+{
+    public class NetTopologySuiteTests : TestBase
+    {
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            using (var conn = OpenConnection())
+            using (var cmd = new NpgsqlCommand("SELECT postgis_version()", conn))
+            {
+                try
+                {
+                    cmd.ExecuteNonQuery();
+                }
+                catch (PostgresException)
+                {
+                    cmd.CommandText = "SELECT version()";
+                    var versionString = (string)cmd.ExecuteScalar();
+                    Debug.Assert(versionString != null);
+                    var m = Regex.Match(versionString, @"^PostgreSQL ([0-9.]+(\w*)?)");
+                    if (!m.Success)
+                        throw new Exception("Couldn't parse PostgreSQL version string: " + versionString);
+                    var version = m.Groups[1].Value;
+                    var prerelease = m.Groups[2].Value;
+                    if (!string.IsNullOrWhiteSpace(prerelease))
+                        Assert.Ignore($"PostGIS not installed, ignoring because we're on a prerelease version of PostgreSQL ({version})");
+                    TestUtil.IgnoreExceptOnBuildServer("PostGIS extension not installed.");
+                }
+            }
+        }
+
+        protected override NpgsqlConnection OpenConnection(string connectionString = null)
+        {
+            NetTopologySuiteBootstrapper.Bootstrap();
```

Comment:
@YohDeadfall, this line seems mandatory, otherwise NTS doesn't work. Is there any reason to force users to do it rather than putting it in the plugin's `UseNetTopologySuite()`? If this is called twice is there any problem or is the 2nd time ignored?

Reply:
Unfortunately, the second call will replace the previous value of `GeometryServiceProvider.Instance` which could be set by an user. There is no check to do nothing if NTS is initialized already. See [the source](https://github.com/NetTopologySuite/NetTopologySuite/blob/af938c521caf2c5bd3ce122ecc9b73b6bf10dad0/NetTopologySuite/NetTopologySuiteBootstrapper.cs#L15-L18). But we can check if `GeometryServiceProvider.Instance` is set and call `Bootstrap` in `UseNetTopologySuite`.

```csharp
public static INpgsqlTypeMapper UseNetTopologySuite(
    this INpgsqlTypeMapper mapper,
    ICoordinateSequenceFactory coordinateSequenceFactory = null,
    IPrecisionModel precisionModel = null,
    Ordinates handleOrdinates = Ordinates.None)
{
    IGeometryService geometryService;
    try
    {
        geometryService = GeometryServiceProvider.Instance;
    }
    catch (InvalidOperationException)
    {
        NetTopologySuiteBootstrapper.Bootstrap();
        geometryService = GeometryServiceProvider.Instance;
    }
    // other stuff
}
```

Yep, it's the check I mentioned. NTS throw an exception if it can't use a reflection based approach to initialize `GeometryServiceProvider` ([source](https://github.com/NetTopologySuite/GeoAPI/blob/134354b16d725f2b5e4d7e3a60abc8afedf089a4/src/GeoAPI/GeometryServiceProvider.cs#L100)). So I think that users which use NTS know about the initialization and will do it by themselves.